gtsfm/bundle/bundle_adjustment.py

```diff
@@ -167,14 +170,25 @@ def get_metrics_from_sfm_data(sfm_data: GtsfmData, suffix: str) -> List[GtsfmMet
         # filter the largest errors
         filtered_result = optimized_data.filter_landmarks(self.output_reproj_error_thresh)
 
-        if wTi_list_gt is not None:
+        if cameras_gt is not None:
             # align the sparse multi-view estimate after BA to the ground truth pose graph.
-            filtered_result = filtered_result.align_via_Sim3_to_poses(wTi_list_gt)
+            filtered_result = filtered_result.align_via_Sim3_to_poses(
+                wTi_list_ref=[cam.pose() for cam in cameras_gt] if cameras_gt else []
```

Comment:
I think you should create a variable for [cam.pose() for cam in cameras_gt]. Improves readability and youre also using it in multiple places.

Reply:
updated